aws_topology/stackstate_checks/aws_topology/resources/ec2.py

```diff
@@ -53,119 +103,163 @@ def process_all(self, filter=None):
             self.process_security_groups()
         if not filter or "vpcs" in filter:
             self.process_vpcs()
+        if not filter or "subnets" in filter:
+            self.process_subnets()
         if not filter or "vpn_gateways" in filter:
             self.process_vpn_gateways()
 
+    @set_required_access_v2("ec2:DescribeInstanceTypes")
+    def collect_instance_types(self):
+        # This list does not change often, safe to hold in memory
```

Comment:
Looks like it indeed make sense to keep this in memory, but I think we should have an expiration for this then? Currently the only way this is emptied is by restarting the agent if I understand this correctly?

Reply:
Good point - I realise that the list can grow, but the items in it never change. I changed this around so I only grab an instance type one at a time, when I need it, then add it to a dict. If it's already in the dict we read it from the dict, otherwise we go and get it from the API. It means that we can always grab new instance types, and speeds up processing as I'm not grabbing the entire list.